app/uk/gov/hmrc/tradergoodsprofilesrouter/models/request/eis/accreditationrequests/RequestAccreditation.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.tradergoodsprofilesrouter.models.request.eis.accreditationrequests
+
+import play.api.libs.functional.syntax.toFunctionalBuilderOps
+import play.api.libs.json.{JsPath, OWrites, Reads}
+import uk.gov.hmrc.tradergoodsprofilesrouter.utils.ValidationSupport.Reads.lengthBetween
+import scala.Function.unlift
+
+case class RequestAccreditation(
+  eori: String,
+  requestorName: String,
+  recordId: String,
+  requestorEmail: String
+)
+
+object RequestAccreditation {
+
+  implicit val reads: Reads[RequestAccreditation] =
+    ((JsPath \ "eori").read(lengthBetween(14, 17)) and
+      (JsPath \ "requestorName").read(lengthBetween(1, 70)) and
+      (JsPath \ "recordId").read(lengthBetween(36, 36)) and
+      (JsPath \ "requestorEmail")
```

Comment:
I think we need a custom email validation here. I'm sure QA will try to break the system by passing invalid email.
You can do something like this
```
      (JsPath \ "requestorEmail")
        .read(lengthBetween(3, 254).keepAnd(validEmailAddress)))(RequestAccreditation.apply _)
```

and add the following to the `ValidationSupport.scala`
```
object ValidationSupport {

private val emailValidator: EmailValidator = EmailValidator.getInstance(true)

def isValidEmailAddress(emailAddress: String): Boolean = emailValidator.isValid(emailAddress)

object Reads {

    val validEmailAddress: Reads[String] = verifying(isValidEmailAddress)
}

}
```